app/src/main/java/it/niedermann/nextcloud/deck/ui/settings/DarkModeSetting.java

```diff
@@ -0,0 +1,66 @@
+package it.niedermann.nextcloud.deck.ui.settings;
+
+import androidx.appcompat.app.AppCompatDelegate;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Possible values of the Dark Mode Setting.
+ * <p>
+ * The Dark Mode Setting can be stored in {@link android.content.SharedPreferences} as String by using {@link DarkModeSetting#name()} and received via {@link DarkModeSetting#valueOf(String)}.
+ * <p>
+ * Additionally, the equivalent {@link AppCompatDelegate}-Mode can be received via {@link #getModeId()}. To convert a {@link AppCompatDelegate}-Mode to a {@link DarkModeSetting}, use {@link #fromModeID(int)}
+ *
+ * @see AppCompatDelegate#MODE_NIGHT_YES
+ * @see AppCompatDelegate#MODE_NIGHT_NO
+ * @see AppCompatDelegate#MODE_NIGHT_FOLLOW_SYSTEM
+ */
+public enum DarkModeSetting {
+    // WARNING - The names of the constants must *NOT* be changed since they are used as keys in SharedPreferences
```

Comment:
Actually i always dislike this approach in the Notes app - the Enum names must magically match the Strings `pref_value_theme_*`.  Maybe we can do better here, what do you think?

We could add a second argument to the constructor `@StringRes int preferenceValue` which points to the corresponding `R.string.pref_value_theme_*`. This way we have a single source of truth for the preference value and could theoretically rename the Enum entries in the future. (though this will be a bit extra work when currently calling `DarkModeSetting#name()`)

Reply:
Done.